src/context.rs

```diff
@@ -25,38 +24,73 @@ impl Default for Match {
     }
 }
 
-pub struct Context<'a> {
-    schema: &'a Schema,
-    values: FnvHashMap<String, Vec<Value>>,
+pub struct Context {
+    values: Vec<Option<Vec<Value>>>,
```

Comment:
uses `values` array to store all field values, `values` index match the index of `Router`'s `fields` array